packages/flow/src/main.rs

```diff
@@ -41,6 +41,126 @@ fn manifest() {
+                        {
+                            "name": "check",
+                            "description": "Specifies the check to do to the value. \
+                            equals: compile the content if the variable equals the given value. \
+                            differs-to: compile the content if the variable differs from the given value. \
+                            defined: compile the content if the variable is defined. \
+                            undefined: compile the content if the variable is undefined.
+                            ",
+                            "default": "defined",
+                            "type": ["equals", "differs-to", "defined", "undefined"]
+                        },
+                        {
+                            "name": "value",
+                            "description": "Specifies the value to compare to",
+                            "default": ""
+                        },
+                        {
+                            "name": "case-sensitive",
+                            "description": "Specifies if 'equals/differs' checks are case sensitive.",
+                            "default": "case-sensitive",
+                            "type": ["case-sensitive", "case-insensitive"]
+                        }
+                    ],
+                    "variables": {
+                        "$constant": {"type": "constant", "access": "read"}
+                    },
+                    "unknown-content": true
+                },
+                {
+                    "from": "if-set",
+                    "to": ["any"],
+                    "description": "Conditionally compile content based on a set. Example: [if-set imports contains-all foo,bar,baz]",
+                    "arguments": [
+                        {
+                            "name": "set",
+                            "description":
+                                "The set to use for conditional compilation"
+                        },
+                        {
+                            "name": "check",
+                            "description": "Specifies the check to do to the value. \
+                            does-not-contain: compile the content if the set does not contain the given string. \
+                            contains: compile the content if the set contains the given string. \
+                            contains-all: compile the content if the set contains all of the comma-separated strings. \
+                            contains-any: compile the content if the set contains any of the comma-separated strings. \
+                            contains-none: compile the content if the set contains none of the comma-separated strings. \
+                            non-empty: compile the content if the set is defined and has at least one string. \
+                            empty-or-undefined: compile the content if the set is undefined or is empty
```

Comment:
as an alternative we could have a `empty` and `undefined` option that both do the same thing if you prefer that.

Reply:
As discussed in private, since a list can't be empty (just undefined), `empty` either has to always fail (which is unexpected) or lie. Additionally it may be confusing to have two options doing the same thing. So I will skip this one